app/assets/javascripts/react_router.js

```diff
@@ -0,0 +1,423 @@
+import UserListView from "admin/views/user/user_list_view";
+import TeamListView from "admin/views/team/team_list_view";
+import TaskListView from "admin/views/task/task_list_view";
+import TaskCollection from "admin/models/task/task_collection";
+import TaskTypeListView from "admin/views/tasktype/task_type_list_view";
+import ProjectListView from "admin/views/project/project_list_view";
+import StatisticView from "admin/views/statistic/statistic_view";
+import WorkloadListView from "admin/views/workload/workload_list_view";
+import WorkloadCollection from "admin/models/workload/workload_collection";
+import ScriptListView from "admin/views/scripts/script_list_view";
+import ProjectCreateView from "admin/views/project/project_create_view";
+import ProjectEditView from "admin/views/project/project_edit_view";
+import ProjectModel from "admin/models/project/project_model";
+import TaskQueryView from "admin/views/task/task_query_view";
+import TaskModel from "admin/models/task/task_model";
+import TaskCreateView from "admin/views/task/task_create_view";
+import TaskCreateFromView from "admin/views/task/task_create_subviews/task_create_from_view";
+import TaskTypeModel from "admin/models/tasktype/task_type_model";
+import TaskTypeCreateView from "admin/views/tasktype/task_type_create_view";
+import ScriptCreateView from "admin/views/scripts/script_create_view";
+import ScriptModel from "admin/models/scripts/script_model";
+import TaskOverviewView from "admin/views/task/task_overview_view";
+import TaskOverviewCollection from "admin/models/task/task_overview_collection";
+import PaginationCollection from "admin/models/pagination_collection";
+
+import type { OxalisState, SkeletonTracingTypeTracingType } from "oxalis/store";
+
+const { Content } = Layout;
+
+type BrowserLocationType = {
+  key: string,
+  pathname: string,
+  search: string,
+  hash: string,
+  state: Object,
+};
+
+type ReactRouterArgumentsType = {
+  location: BrowserLocationType,
+  match: {
+    params: { [string]: string },
+    isExact: boolean,
+    path: string,
+    url: string,
+  },
+  history: {
+    length: number,
+    action: string,
+    location: BrowserLocationType,
+    pathname: string,
+    search: string,
+    hash: string,
+    state: string,
+    push: (string, [Object]) => void,
+    replace: (string, [Object]) => void,
+    go: number => void,
+    goBack: () => void,
+    goForward: () => void,
+    block: string => string,
+  },
+};
+
+// <Redirect
+// to={{
+// pathname: "/login",
+// state: { from: props.location },
+// }}
+// />
+const SecuredRoute = ({ component: Component, render, isAuthenticated, ...rest }: Object) => (
+  <Route
+    {...rest}
+    render={props => {
+      if (isAuthenticated) {
+        return Component ? <Component {...props} /> : render(props);
+      } else {
+        return <LoginView />;
+      }
+    }}
+  />
+);
+
+class ReactRouter extends React.Component<*> {
```

Comment:
Do we still have code splitting?

Reply:
Nope. 
![screen shot 2017-10-16 at 17 17 06](https://user-images.githubusercontent.com/1105056/31619802-e072cd76-b295-11e7-8b82-b8b93e4f7cfc.png)